In [2]:
import pandas as pd

---

# 1. 데이터 이해

In [3]:
df = pd.read_csv('../../data/customer_hm.csv')
df.shape

(1048575, 6)

In [ ]:
print('# 컬럼명')
print(df.columns)
print('-----------'*5)
print('# 데이터 타입 및 결측지')
print(df.info())

In [ ]:
df.head()

In [ ]:
for col in df.columns:
    if col not in ['customer_id', 'age']:
        print(f"\n{col}")
        print(df[col].unique())

## →
- 행: 1048575
- 열: 6
- customers_hm 컬럼 설명
    - `customer_id`: 고객 id
    - `FN`: 패션 뉴스 구독 여부
    - `Active`: 커뮤니케이션 가능 여부
    - `club_member_status`: 회원 상태(신규, 활성, 탈퇴)
    - `fashion_news_frequency`: 패션 뉴스 알림 주기
    - `age`: 나이
- `fashion_news_frequency` 컬럼 결측치 1개

<br>

Next
- `Active` 컬럼은 뭐지?
    - `Active`와 `fashon_new_frequency`의 관계 확인

---

# 2. 데이터 전처리

## 2-1. 결측치 처리

### 2-1-1. 결측치 확인

In [ ]:
df.isna().sum()

In [ ]:
# 빈 문자열 및 문자열로 된 빈 값 확인
print((df == '').sum())
print('----------'*5)
print(df.isin(['None', 'NULL', 'null']).sum())

In [ ]:
df[df.isnull().any(axis=1)]

### 2-1-2. 결측치 처리

In [5]:
df[(df['FN'] == 1) & (df['Active'] == 0)
][['FN', 'Active', 'club_member_status', 'fashion_news_frequency']]

FN  Active club_member_status fashion_news_frequency
133       1       0             ACTIVE              Regularly
241       1       0             ACTIVE              Regularly
326       1       0             ACTIVE                   NONE
426       1       0             ACTIVE              Regularly
466       1       0             ACTIVE              Regularly
...      ..     ...                ...                    ...
1048188   1       0             ACTIVE              Regularly
1048217   1       0             ACTIVE              Regularly
1048266   1       0             ACTIVE              Regularly
1048294   1       0             ACTIVE              Regularly
1048510   1       0         PRE-CREATE              Regularly

[9864 rows x 4 columns]

In [4]:
df[(df['FN'] == 1) & (df['Active'] == 0)
]['fashion_news_frequency'].value_counts(dropna=False)

fashion_news_frequency
Regularly    9598
NONE          238
Monthly        27
NaN             1
Name: count, dtype: int64

In [ ]:
df_clean = df.copy()

df_clean.loc[876108, 'fashion_news_frequency'] = 'Regularly'
df_clean.loc[876108]

→ <br>
`FN`: 1 이고 `Active`: 0 인 데이터 중에서 `fashion_news_frequency`의 최빈값이 `Regularly`로 확인되어, `fashion_news_frequency`에 결측치가 존재하는 인덱스 번호 `876108`의 `fashion_news_frequency` 값을 `Regularly`로 수정
